### ISSUE TITLE
activation function at last layer

### ISSUE THREAD

**Issue by tingjhenjiang** [GarrettJenkinson/condor_tensorflow#1]
Hello,
I've a dataset in which the labels are like (0,1,2,3). It means the number of classes in Y is 4.

Method 1:

Using the `condor.CondorOrdinalEncoder(nclasses=4).fit_transform(labels)` to transform labels to an array in shape (n, 3).
[ [0,0,1],
　[1,0,0] ] as model prediction objects.
The last layer is tf.keras.layers.Dense(units=4-1), according to [the readme](https://pythonrepo.com/repo/garrettjenkinson-condor_tensorflow-python-deep-learning), however by this design the default activation function of the last layer would be None/Linear( f(x) = x), and the output of the model would be simple logits.
Should I keep the model outputs simple logits(no activation function)?

Method 2:

If I use `tf.keras.layers.Dense(units=4-2, activation=condor.ordinal_softmax)` as the last layer together with label data in shape (n, 3), would that be fine?
(the condor.ordinal_softmax function would increase the number of dimension)

Method 3:
Or I should use `tf.keras.layers.Dense(units=4-1, activation=condor.ordinal_softmax)` as the last layer together with label data in shape (n, 4)?

Which method is better?
Thank you!

**Comment by GarrettJenkinson** [comment]
Hi thanks for you interest in the method. At this time, a good demo of the various usage patterns can be found here:

https://github.com/GarrettJenkinson/condor_tensorflow/blob/main/docs/CONDOR_TensorFlow_demo.ipynb

We generally suggest using logits as the output of the network (i.e., a dense layer output with no activations and N-1 neurons where N is number of ordinal labels), and then as needed use the activation function to convert to probabilities. See the section titled "Cumulative logits to probabilities" in the notebook for details of working with the logit outputs. 

Regarding label encoding, see the section of the ipynb titled "Universal Sentence Encoder model (CONDOR applied with minimal code changes)" versus "Universal Sentence Encoder model (using pre-encoded labels for faster processing)". Basically, the code will be simpler if you just leave your labels (0,1,2,3), but it will run faster if you pre-encode into the binary vectors. 

Let me know if you have further questions or if anything remains unclear.

**Comment by GarrettJenkinson** [comment]
I am going to close this issue, but if you have noticed any errors in the documentation let me know and I will correct them.
